modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java

```diff
@@ -139,6 +140,92 @@ public static Object resolveCursor(Object value) {
     return value;
   }
 
+  /**
+   * Obtains a {@link CursorStream} that cannot be closed based on the {@code value}, if one is available.
+   *
+   * @return the given {@code value}, but wrapped in a way that it is not posible to close it.
+   */
+  public static Object resolveToUnclosableCursor(CursorStream value) {
+    return new UnclosableCursorStream(value);
+  }
+
+  private static class UnclosableCursorStream extends CursorStream {
```

Comment:
I would extract this from
this class,  and remove the static method thst creates instances of it.

This could probably live in other place